dev-php/symfony-dependency-injection/symfony-dependency-injection-2.1.0.ebuild

```diff
@@ -0,0 +1,35 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+DESCRIPTION="Symfony DependencyInjection Component"
+HOMEPAGE="https://github.com/symfony/dependency-injection"
+
+SRC_URI="https://github.com/symfony/dependency-injection/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+S="${WORKDIR}/dependency-injection-${PV}"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+RDEPEND="dev-lang/php:*
+	dev-php/fedora-autoloader
+	=dev-php/symfony-yaml-2.1.0
+	=dev-php/symfony-config-2.1.0"
+DEPEND="${RDEPEND}"
+
+src_install() {
+	# I've kept the same path name that Fedora use
+	insinto "/usr/share/php/Symfony/DependencyInjection"
+	doins -r *
+	doins "${FILESDIR}"/autoload.php
+	dodoc README.md
+}
+
+src_test() {
+	phpunit --bootstrap /usr/share/php/Symfony/DependencyInjection/autoload.php || die "test suite failed"
```

Comment:
Ok I did it a cleaner than what I said, I added a src_prepare to copy the autoloader file from the FILESDIR in the WORKDIR, like
```
RDEPEND="dev-lang/php:*
        =dev-php/fedora-autoloader-0.2.1"
DEPEND="test? ( ${RDEPEND}
                                dev-php/phpunit )"

src_prepare() {
        default
        if use test; then
                cp "${FILESDIR}"/autoload.php "${S}"/autoload-test.php
                sed -i -e "s:__DIR__:'${S}/src/Seld/JsonLint':" "${S}"/autoload-test.php
        fi
}

src_install() {
        insinto "/usr/share/php/Seld/JsonLint"
        doins -r src/Seld/JsonLint/. "${FILESDIR}"/autoload.php
        dodoc README.mdown
}

src_test() {
        phpunit --bootstrap "${S}"/autoload-test.php || die "test suite failed"
}
```
After I change the --bootstrap to target this new file, so now the test works when not installed,
so I can install the composer with FEATURES=test and it pull everything and tests works
```
FEATURES=test emerge -av "=dev-php/composer-1.2.2"
 * IMPORTANT: 3 news items need reading for repository 'gentoo'.
 * Use eselect news read to view new items.


These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild  N     ] dev-php/phpunit-4.3.1::gentoo  0 KiB
[ebuild  N    #] dev-php/fedora-autoloader-0.2.1::GuillaumeSeren  USE="{test}" 0 KiB
[ebuild  N    #] dev-php/fig-log-1.0.2::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/symfony-process-2.8.12::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/semver-1.0.0::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/spdx-licenses-1.0.0::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/symfony-finder-2.7.20::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/jsonlint-1.4.0::GuillaumeSeren  USE="{test}" 0 KiB
[ebuild  N    #] dev-php/symfony-filesystem-2.7.20::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/json-schema-2.0.0::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/phar-utils-1.0.0::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/ca-bundle-1.0.0::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/cli-prompt-1.0.0::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/symfony-yaml-2.1.0::GuillaumeSeren  USE="{test}" 0 KiB
[ebuild  N    #] dev-php/symfony-config-2.1.0::GuillaumeSeren  USE="{test}" 0 KiB
[ebuild  N    #] dev-php/symfony-dependency-injection-2.1.0::GuillaumeSeren  USE="{test}" 0 KiB
[ebuild  N    #] dev-php/symfony-event-dispatcher-2.1.0::GuillaumeSeren  USE="{test}" 0 KiB
[ebuild  N    #] dev-php/symfony-console-2.7.9::GuillaumeSeren  0 KiB
[ebuild  N    #] dev-php/composer-1.2.2::GuillaumeSeren  0 KiB

Total: 19 packages (19 new), Size of downloads: 0 KiB

Would you like to merge these packages? [Yes/No] Yes
>>> Verifying ebuild manifests
>>> Emerging (1 of 19) dev-php/phpunit-4.3.1::gentoo
>>> Installing (1 of 19) dev-php/phpunit-4.3.1::gentoo
>>> Emerging (2 of 19) dev-php/fedora-autoloader-0.2.1::GuillaumeSeren
>>> Installing (2 of 19) dev-php/fedora-autoloader-0.2.1::GuillaumeSeren
>>> Emerging (3 of 19) dev-php/fig-log-1.0.2::GuillaumeSeren
>>> Installing (3 of 19) dev-php/fig-log-1.0.2::GuillaumeSeren
>>> Emerging (4 of 19) dev-php/symfony-process-2.8.12::GuillaumeSeren
>>> Installing (4 of 19) dev-php/symfony-process-2.8.12::GuillaumeSeren
>>> Emerging (5 of 19) dev-php/semver-1.0.0::GuillaumeSeren
>>> Installing (5 of 19) dev-php/semver-1.0.0::GuillaumeSeren
>>> Emerging (6 of 19) dev-php/spdx-licenses-1.0.0::GuillaumeSeren
>>> Installing (6 of 19) dev-php/spdx-licenses-1.0.0::GuillaumeSeren
>>> Emerging (7 of 19) dev-php/symfony-finder-2.7.20::GuillaumeSeren
>>> Installing (7 of 19) dev-php/symfony-finder-2.7.20::GuillaumeSeren
>>> Emerging (8 of 19) dev-php/jsonlint-1.4.0::GuillaumeSeren
>>> Installing (8 of 19) dev-php/jsonlint-1.4.0::GuillaumeSeren
>>> Emerging (9 of 19) dev-php/symfony-filesystem-2.7.20::GuillaumeSeren
>>> Installing (9 of 19) dev-php/symfony-filesystem-2.7.20::GuillaumeSeren
>>> Emerging (10 of 19) dev-php/json-schema-2.0.0::GuillaumeSeren
>>> Installing (10 of 19) dev-php/json-schema-2.0.0::GuillaumeSeren
>>> Emerging (11 of 19) dev-php/phar-utils-1.0.0::GuillaumeSeren
>>> Installing (11 of 19) dev-php/phar-utils-1.0.0::GuillaumeSeren
>>> Emerging (12 of 19) dev-php/ca-bundle-1.0.0::GuillaumeSeren
>>> Installing (12 of 19) dev-php/ca-bundle-1.0.0::GuillaumeSeren
>>> Emerging (13 of 19) dev-php/cli-prompt-1.0.0::GuillaumeSeren
>>> Installing (13 of 19) dev-php/cli-prompt-1.0.0::GuillaumeSeren
>>> Emerging (14 of 19) dev-php/symfony-yaml-2.1.0::GuillaumeSeren
>>> Installing (14 of 19) dev-php/symfony-yaml-2.1.0::GuillaumeSeren
>>> Emerging (15 of 19) dev-php/symfony-config-2.1.0::GuillaumeSeren
>>> Installing (15 of 19) dev-php/symfony-config-2.1.0::GuillaumeSeren
>>> Emerging (16 of 19) dev-php/symfony-dependency-injection-2.1.0::GuillaumeSeren
>>> Installing (16 of 19) dev-php/symfony-dependency-injection-2.1.0::GuillaumeSeren
>>> Emerging (17 of 19) dev-php/symfony-event-dispatcher-2.1.0::GuillaumeSeren
>>> Installing (17 of 19) dev-php/symfony-event-dispatcher-2.1.0::GuillaumeSeren
>>> Emerging (18 of 19) dev-php/symfony-console-2.7.9::GuillaumeSeren
>>> Installing (18 of 19) dev-php/symfony-console-2.7.9::GuillaumeSeren
>>> Emerging (19 of 19) dev-php/composer-1.2.2::GuillaumeSeren
>>> Installing (19 of 19) dev-php/composer-1.2.2::GuillaumeSeren
>>> Recording dev-php/composer in "world" favorites file...
>>> Jobs: 19 of 19 complete                         Load avg: 4.12, 4.17, 4.36

 * Messages for package dev-php/phpunit-4.3.1:

 * phpunit can optionally use json, pdo-sqlite and pdo-mysql features.
 * If you want those, emerge dev-lang/php with USE="json pdo sqlite mysql".
>>> Auto-cleaning packages...

```